kahuna/public/js/search/query.js

```diff
@@ -170,12 +175,27 @@ query.controller('SearchQueryCtrl', [
     // we can't user dynamic values in the ng:true-value see:
     // https://docs.angularjs.org/error/ngModel/constexpr
     mediaApi.getSession().then(session => {
+        const isNonFree = storage.getJs("isNonFree");
+        const isUploadedByMe = storage.getJs("isUploadedByMe");
+
         ctrl.user = session.user;
-        ctrl.filter.uploadedByMe = ctrl.uploadedBy === ctrl.user.email;
+        if (isUploadedByMe === undefined) {
+              ctrl.filter.uploadedByMe = ctrl.uploadedBy === ctrl.user.email;
+              storage.setJs("isUploadedByMe",ctrl.filter.uploadedByMe);
+        }
+        else {
+          ctrl.filter.uploadedByMe =  isUploadedByMe;
+        }
 
-        if (ctrl.filter.nonFree === undefined) {
+        if (isNonFree === undefined) {
           ctrl.filter.nonFree = session.user.permissions.showPaid ?
             session.user.permissions.showPaid : undefined;
+            storage.setJs("isNonFree", session.user.permissions.showPaid ? session.user.permissions.showPaid : false);
+        }
+        else if (isNonFree === true || isNonFree === "true") {
```

Comment:
non-blocking: IIUC this could be more concisely expressed as `Boolean(isNonFree)`.

Reply:
@jonathonherbert it appears @abdelrahmansd's way is safer, `Boolean` constructor will coerce any truthy value to true

![image](https://user-images.githubusercontent.com/19289579/160790643-dc5a4985-4714-47dd-a2e8-397bc47f2662.png)
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean/Boolean

Reply:
Good spot, thanks for the correction!